src/request.rs

```diff
@@ -800,6 +828,31 @@ impl Builder {
         self
     }
 
+    /// Get the URI for this request
+    /// 
+    /// By default this is `/`
+    /// # Examples
+    /// 
+    /// ```
+    /// # use http::*;
+    /// 
+    /// let mut req = Request::builder();
+    /// assert_eq!(req.get_uri().unwrap().to_string(), "/" );
+    /// req.uri("https://www.rust-lang.org/");
+    /// assert_eq!(req.get_uri().unwrap().to_string(), "https://www.rust-lang.org/" );
+    /// ```
+    pub fn get_uri(&self) -> Option<Uri>
```

Comment:
This should return an `Option<&Uri>`. The user can decide if they need to clone it, but if not, no need to be wasteful :)